worlds/civ_6/Items.py

```diff
@@ -0,0 +1,302 @@
+    "TECH_BALLISTICS",
+    "TECH_STEAM_POWER",
+    "TECH_SANITATION",
+    "TECH_COMPUTERS",
+    "TECH_COMBUSTION",
+    "TECH_TELECOMMUNICATIONS",
+    "TECH_ROBOTICS",
+    "CIVIC_FEUDALISM",
+    "CIVIC_GUILDS",
+    "CIVIC_THE_ENLIGHTENMENT",
+    "CIVIC_MERCANTILISM",
+    "CIVIC_CONSERVATION",
+    "CIVIC_CIVIL_SERVICE",
+    "CIVIC_GLOBALIZATION",
+    "CIVIC_COLD_WAR",
+    "CIVIC_URBANIZATION",
+    "CIVIC_NATIONALISM",
+    "CIVIC_MOBILIZATION",
+    "PROGRESSIVE_NEIGHBORHOOD",
+    "PROGRESSIVE_PRESERVE"
+]
+
+
+class FillerItemRarity(Enum):
+    COMMON = "COMMON"
+    UNCOMMON = "UNCOMMON"
+    RARE = "RARE"
+
+
+FILLER_DISTRIBUTION: Dict[FillerItemRarity, float] = {
+    FillerItemRarity.RARE: 0.025,
+    FillerItemRarity.UNCOMMON: 0.2,
+    FillerItemRarity.COMMON: 0.775,
+}
+
+
+class FillerItemData:
+    name: str
+    type: str
+    rarity: FillerItemRarity
+    civ_name: str
+
+    def __init__(self, data: GoodyHutRewardData):
+        self.name = data["Name"]
+        self.rarity = FillerItemRarity(data["Rarity"])
+        self.civ_name = data["Type"]
+
+
+filler_data: Dict[str, FillerItemData] = {item["Name"]: FillerItemData(item) for item in get_goody_hut_rewards_data()}
+
+
+class CivVIItemData:
+    civ_vi_id: int
+    classification: ItemClassification
+    name: str
+    code: int
+    cost: int
+    item_type: CivVICheckType
+    progression_name: Optional[str]
+    civ_name: Optional[str]
+    era: Optional[EraType]
+
+    def __init__(self, name: str, civ_vi_id: int, cost: int, item_type: CivVICheckType, id_offset: int, classification: ItemClassification, progression_name: Optional[str], civ_name: Optional[str] = None, era: Optional[EraType] = None):
+        self.classification = classification
+        self.civ_vi_id = civ_vi_id
+        self.name = name
+        self.code = civ_vi_id + CIV_VI_AP_ITEM_ID_BASE + id_offset
+        self.cost = cost
+        self.item_type = item_type
+        self.progression_name = progression_name
+        self.civ_name = civ_name
+        self.era = era
+
+
+class CivVIEvent(Item):
+    game: str = "Civilization VI"
+
+
+class CivVIItem(Item):
+    game: str = "Civilization VI"
+    civ_vi_id: int
+    item_type: CivVICheckType
+
+    def __init__(self, item: CivVIItemData, player: int, classification: Optional[ItemClassification] = None):
+        super().__init__(item.name, classification or item.classification, item.code, player)
+        self.civ_vi_id = item.civ_vi_id
+        self.item_type = item.item_type
+
+
+def format_item_name(name: str) -> str:
+    name_parts = name.split("_")
+    return " ".join([part.capitalize() for part in name_parts])
+
+
+def get_item_by_civ_name(item_name: str, item_table: Dict[str, 'CivVIItemData']) -> 'CivVIItemData':
+    """Gets the names of the items in the item_table"""
+    global _items_by_civ_name
+    if not _items_by_civ_name:
+        _items_by_civ_name = {item.civ_name: item for item in item_table.values() if item.civ_name}

```

Comment:
I feel like there's a better way to do this but it's hard to say. Like maybe there's a good way to throw the world class in here so that _items_by_civ_name can be on that instead of a global, but ehhh it's probably fine

Reply:
Yeah I went back and forth on this, with it being a static list just based on the data in those JSON files I like not needing the reference to the world in order to use it in other areas (this is used in the client as well where the world doesn't exist 🤷)